Apps/CommonData/src/ErrorHandling/ProblemDetailsException.cs

```diff
@@ -0,0 +1,80 @@
+// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Common.Data.ErrorHandling
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// Represents a custom api patient exception.
+    /// </summary>
+    [Serializable]
+    public class ProblemDetailsException : Exception
```

Comment:
I don't think there should be an exception for this, the idea of problemDetail is to be able to map native exceptions or API specific states to a machine readable/actionable state.

Reply:
This is used for custom thrown exceptions and is handled in the HttpResponseExceptionFilter.  It checks the exception thrown and if applicable, it maps the contents of the exception to Problem Details and adjusts the response code accordingly.

Reply:
Yes, I understand. My question was why not let the exceptions bubble to the filter and then map to ProblemDetail?

Reply:
1. Exceptions that bubble to the top and not intercepted in the filter, will be a 500
2. These are custom validation exceptions.  In client registry, they check for specific results in regards to codes sent back in the soap message. In the past they used the ResultError inside the RequestResult object.  Now a custom exception is thrown with  response code that can be specified like NOT FOUND, etc instead of relying on a 500 Internal Sever Error.